src/ray/raylet/node_manager.cc

```diff
@@ -3113,6 +3100,86 @@ void NodeManager::ReportWorkerOOMKillStats() {
   number_workers_killed_ = 0;
 }
 
+std::shared_ptr<AgentManager> NodeManager::CreateDashboardAgentManager(
+    const NodeID &self_node_id, const NodeManagerConfig &config) {
+  auto agent_command_line = ParseCommandLine(config.dashboard_agent_command);
+
+  if (agent_command_line.empty()) {
+    return nullptr;
+  }
+
+  for (auto &arg : agent_command_line) {
+    auto node_manager_port_position = arg.find(kNodeManagerPortPlaceholder);
+    if (node_manager_port_position != std::string::npos) {
+      arg.replace(node_manager_port_position,
+                  strlen(kNodeManagerPortPlaceholder),
+                  std::to_string(GetServerPort()));
+    }
+  }
+  // Disable metrics report if needed.
+  if (!RayConfig::instance().enable_metrics_collection()) {
+    agent_command_line.push_back("--disable-metrics-collection");
+  }
+
+  // Create a non-zero random agent_id to pass to the child process
+  // We cannot use pid an id because os.getpid() from the python process is not
+  // reliable when using a launcher.
+  // See https://github.com/ray-project/ray/issues/24361 and Python issue
+  // https://github.com/python/cpython/issues/83086
+  int agent_id = 0;
```

Comment:
can you double check if we don't need this for runtime env agent?

Reply:
I privately pinged the original author and cc'ed you

Reply:
I touched this in #24968. The `agent_id` is used to verify that the agent starts, since there can be two processes with two PIDs. For more information about this see the PR description, which shows how starting python actually starts two processes when inside a virtualenv on windows.

Testing all this on windows was enabled in #23613, so if windows tests pass here there should be some confidence that the use of `agent_id` is working correctly.

Reply:
will take a look on the windows tests.